src/NuGet.Core/NuGet.Common/Telemetry/INuGetTelemetryService.cs

```diff
@@ -1,13 +1,20 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-
 using System;
 
 namespace NuGet.Common
 {
+    /// <summary> Abstraction of NuGet telemetry service. </summary>
     public interface INuGetTelemetryService
     {
+        /// <summary> Send a <see cref="TelemetryEvent"/> to VS telemetry. </summary>
+        /// <param name="telemetryData"> Telemetry event to send. </param>
         void EmitTelemetryEvent(TelemetryEvent telemetryData);
+
+        /// <summary> Log a start of telemetry activity to the event log. </summary>
+        /// <param name="activityName"> Name of telemetry activity to log. </param>
+        /// <returns> <see cref="IDisposable"/> which will log end activity marker. </returns>
+        IDisposable StartActivity(string activityName);
```

Comment:
I have no idea if VS for Mac, or JetBrains Rider, or anyone else, implements this interface in their own product. But if they do, adding this method breaks them if they upgrade a a package with this change.

I haven't thought about this PR enough to offer suggestions on how a breaking API can be avoided (in general we only need to avoid breaking API changes in `src\NuGet.Core\*`, not `src\NuGet.Clients\*`). if you have ideas, I encourage you to take that route. Otherwise, I don't imagine many customers (internal or external) implement this interface, so not a big deal to break.

Reply:
I searched github.com and a bit of Internet, and only implementation of this interface is [NuGetVSTelemetryService](https://github.com/WeilerWebServices/NuGet/blob/c21c720b61ba0e61259880a4749d7dd5926b505c/NuGet.Client/src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/NuGetVSActionTelemetryService.cs) on NuGet.VisualStudio.Common.dll.

Reply:
@NuGet/nuget-client This talk about breaking changes in NuGet.Core, made me look... It seems that too much `public` classes/members which should not be public in the first place... Any plans on how to reduce that public surface?

Reply:
Nothing official, but I spent some effort on creating a tool that would analyze assemblies that we have access to, and check what APIs they're actually using: https://github.com/NuGet/Entropy/tree/master/nuget-sdk-usage

While we could never be sure we found every API used, it could give us a good hint. I don't know if this tool will give the team enough confidence to accept reducing public API surface, however.

Reply:
Just to add to @zivkan's point. We've spent plenty of time discussing and I think I speak for both me & @zivkan when I say we've been advocates for being super careful with any new public surface area we define. 

@zivkan did this not too long ago: https://github.com/NuGet/NuGet.Client/pull/3361#issuecomment-618640583